### PR TITLE
Switch to non-parallel stream in REST pagination mapping (Fixes #6803)

### LINKED ISSUE #6803: Switch to non parallel stream in rest pagination mapping
Update needs to be done here: https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/server/pagination/rest/main/RestAppPage.java.mustache

### PR DESCRIPTION
**Pull request by jayasekharreddy-telluri**
Switch to non-parallel stream in REST pagination mapping

Refactor the pagination mapping code in order to switch from using parallelStream() to stream() for processing the data. This change ensures sequential processing of the stream, addressing potential thread-safety concerns and improving the overall reliability of the pagination functionality.

Fixes #6803

### CODE REVIEW

File: src/main/resources/generator/server/pagination/rest/main/RestAppPage.java.mustache [hunk 1]
```diff
@@ -34,7 +34,7 @@ public class Rest{{baseName}}Page<T> {
     Assert.notNull("source", source);
     Assert.notNull("mapper", mapper);
 
-    return new Rest{{baseName}}PageBuilder<>(source.content().parallelStream().map(mapper).toList())
+    return new Rest{{baseName}}PageBuilder<>(source.content().stream().collect(Collectors.toList()))
```
Comment (DamnClin) [inline src/main/resources/generator/server/pagination/rest/main/RestAppPage.java.mustache:37]: You changed behavior here, you need to do `source.content().stream().map(mapper).toList()` and, as murdos pointed out, same thing need to be done in jpa pagination

**Comment by DamnClin** [comment]
The generated code won't compile so I'm not starting the pipeline here :)

File: src/main/resources/generator/server/pagination/rest/main/RestAppPage.java.mustache (hunk 1 above)
Comment (jayasekharreddy-telluri) [reply src/main/resources/generator/server/pagination/rest/main/RestAppPage.java.mustache:37]: i didnt found the jpa pagination class can you help me

Comment (DamnClin) [reply src/main/resources/generator/server/pagination/rest/main/RestAppPage.java.mustache:37]: It is murdos first comment on the issue...

File: src/main/resources/generator/server/pagination/rest/main/RestAppPage.java.mustache [hunk 2]
```diff
@@ -34,7 +34,11 @@ public class Rest{{baseName}}Page<T> {
     Assert.notNull("source", source);
     Assert.notNull("mapper", mapper);
 
+<<<<<<< HEAD
```
Comment (DamnClin) [inline src/main/resources/generator/server/pagination/rest/main/RestAppPage.java.mustache:37]: You have a conflict to fix here

**Comment by jayasekharreddy-telluri** [comment]
hi DamnClin can you check once my latest commit and reveiw it

**Comment by DamnClin** [comment]
@jayasekharreddy-telluri there is a conflict in the branch & it's still draft, we can't merge it as is :)

**Comment by DamnClin** [comment]
Seems @pascalgrimaud can merge it, we'll do it, thanks for your contribution :)

**Comment by pascalgrimaud** [comment]
Thanks and congrats for your 1st contribution @jayasekharreddy-telluri

**Comment by jayasekharreddy-telluri** [comment]
@DamnClin  And @pascalgrimaud  thanks to both of you for your appreciation, and I'm looking forward to assisting you with your next task! Have a fantastic day!

**Pull request merged**